### ISSUE TITLE
Node refuses to shutdown with 1.0.2 RC

### ISSUE THREAD

**Issue by vladzcloudius** [scylladb/scylla#1226]
## Installation details
Scylla version (or git commit hash): 60307f6
Cluster size: 5
OS (RHEL/CentOS/Ubuntu/AWS AMI): Debian jessie

## Hardware details (for performance issues)
Platform (physical/VM/cloud instance type/docker): Amazon c3.8xlarge instance

## Description
I accidentally found this issue when I wanted to restart the test.
When jepsen environment starts a new test it first kills all `scylla`s and `java`s.
So, suddenly two (n1 and n4) refused to shutdown and this is what I saw in their logs:

Log of n4 (192.168.122.14):
```
INFO  2016-04-28 13:44:55,046 [shard 0] storage_service - Drain on shutdown: starts
INFO  2016-04-28 13:44:55,046 [shard 0] gossip - My status = NORMAL
INFO  2016-04-28 13:44:55,046 [shard 0] gossip - Announcing shutdown
INFO  2016-04-28 13:44:55,046 [shard 0] storage_service - Node 192.168.122.14 state jump to normal
INFO  2016-04-28 13:44:55,241 [shard 0] gossip - do_gossip_to_live_member: Favor newly added node 192.168.122.13
WARN  2016-04-28 13:44:56,271 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,271 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,271 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,271 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,271 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,271 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,271 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,271 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,367 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,367 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,367 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,367 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,367 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,367 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,367 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,367 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,367 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:56,368 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
INFO  2016-04-28 13:44:57,081 [shard 0] storage_service - Drain on shutdown: stop_gossiping done
INFO  2016-04-28 13:44:57,081 [shard 0] storage_service - CQL server stopped
INFO  2016-04-28 13:44:57,081 [shard 0] storage_service - Drain on shutdown: shutdown rpc and cql server done
INFO  2016-04-28 13:44:57,082 [shard 0] rpc - client 192.168.122.13: server connection dropped: read: Connection reset by peer
INFO  2016-04-28 13:44:57,082 [shard 3] rpc - client 192.168.122.12: server connection dropped: read: Connection reset by peer
INFO  2016-04-28 13:44:57,082 [shard 2] rpc - client 192.168.122.13: server connection dropped: read: Connection reset by peer
INFO  2016-04-28 13:44:57,082 [shard 2] rpc - client 192.168.122.12: server connection dropped: read: Connection reset by peer
INFO  2016-04-28 13:44:57,082 [shard 2] rpc - client 192.168.122.13: server connection dropped: read: Connection reset by peer
WARN  2016-04-28 13:44:57,241 [shard 0] gossip - Fail to send EchoMessage to 192.168.122.11:0: rpc::timeout_error (rpc call timed out)
WARN  2016-04-28 13:44:57,241 [shard 0] gossip - Fail to send GossipDigestACK2 to 192.168.122.13:0: rpc::closed_error (connection is closed)
```
Log of n1 (192.168.122.11):
```
INFO  2016-04-28 13:44:55,044 [shard 0] storage_service - Drain on shutdown: starts
INFO  2016-04-28 13:44:55,045 [shard 0] gossip - My status = NORMAL
INFO  2016-04-28 13:44:55,045 [shard 0] gossip - Announcing shutdown
INFO  2016-04-28 13:44:55,045 [shard 0] storage_service - Node 192.168.122.11 state jump to normal
INFO  2016-04-28 13:44:57,081 [shard 0] storage_service - Drain on shutdown: stop_gossiping done
INFO  2016-04-28 13:44:57,081 [shard 0] storage_service - CQL server stopped
INFO  2016-04-28 13:44:57,081 [shard 0] storage_service - Drain on shutdown: shutdown rpc and cql server done
INFO  2016-04-28 13:44:57,081 [shard 0] rpc - client 192.168.122.13: server connection dropped: read: Connection reset by peer
INFO  2016-04-28 13:44:57,081 [shard 2] rpc - client 192.168.122.12: server connection dropped: read: Connection reset by peer
INFO  2016-04-28 13:45:03,610 [shard 0] rpc - client 192.168.122.12: fail to connect: Connection refused
```
Only `pkill -9 scylla` killed them.
It doesn't seem like an expected behavior to stuck like that.

**Comment by tzach** [comment]
@vladzcloudius is this an old issue or regression from 1.0.1?

**Comment by vladzcloudius** [comment]
@tzach No idea. This just accidentally happen. This wasn't a part of a test.

**Comment by tzach** [comment]
@vladzcloudius did this ever happen again?
Either we investigate further, or it is not really a High bug.

**Comment by vladzcloudius** [comment]
IIRC this was a bug in a gossiper or in a streaming. @asias do u remember this? If yes, could u send a specific patch that fixed that?

@tzach The bottom line - this didn't happen the last time I ran jepsen tests (for 1.1 IIRC).

**Comment by asias** [comment]
I think the problem was messaging server hanged during shutdown since "Drain on shutdown: shutdown messaging_service done" message was not shown. We have patches in this area, I think we can close the issue now.

**Comment by tzach** [comment]
@asias do you mean b2f2c1c28c7004caae464fcd4dcde12bb5c8c387 ?

I will take your word and close this. We can reopen if it ever happened again

**Comment by asias** [comment]
On Tue, Jun 7, 2016 at 7:54 PM, Tzach Livyatan <notifications@github.com>
wrote:

> @asias <https://github.com/asias> do you mean b2f2c1c
> <https://github.com/scylladb/scylla/commit/b2f2c1c28c7004caae464fcd4dcde12bb5c8c387>
> ?
>
> I will take your word and close this. We can reopen if it ever happened
> again
>

I meant commit like:

commit 357c91a0761efd29ae06730497e4649236861a77
Author: Gleb Natapov <gleb@scylladb.com>
Date:   Tue Mar 22 11:37:31 2016 +0200

    messaging: do not delete client during messaging service shutdown

    Messaging service stop() method calls stop() on all clients. If
    remove_rpc_client_one() is called while those stops are running
    client::stop() will be called twice which not suppose to happen. Fix it
    by ignoring client remove request during messaging service shutdown.

    Fixes #1059


> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scylladb/scylla/issues/1226#issuecomment-224258037>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AATiTBxIIOAQrzsBEN5MJmu5eOqc0Ixsks5qJVvogaJpZM4IR84z>
> .
>



-- 
Asias
